rdt-open-reader/src/main/java/io/ona/rdt/activity/PatientProfileActivity.java

```diff
@@ -65,10 +65,7 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
             // for exceptions thrown and caught during widget instantiation
             Bundle bundle = data.getExtras();
             throw  (RuntimeException) bundle.getSerializable(JsonFormConstants.RESULT_INTENT.RUNTIME_EXCEPTION);
-        } else if (requestCode == REQUEST_CODE_GET_JSON && resultCode == RESULT_CANCELED) {
```

Comment:
This is required to close out the app when the jsonformactivity crashes.

Reply:
Also, how is this changed related to #950?

Reply:
> This is required to close out the app when the jsonformactivity crashes.

I have checked the toast showing if we do not allow the location permission.

> Also, how is this changed related to #950?

yussin has shared the log https://github.com/onaio/rdt-standard/issues/950#issuecomment-816337260 and in the log the is related to this line.

Reply:
this line should only run if the result code is `RESULT_CANCELED`. If permissions are rejected, could you set the result code to something else?

Reply:
I think we don't need to check something else because toast is already showing.

Reply:
What toast are you referring to?

Reply:
my suggestion is to change this line https://github.com/onaio/rdt-standard/blob/a546608c8d1fb5f101d3d175231cfb1af2afb716/rdt-open-reader/src/main/java/io/ona/rdt/activity/RDTJsonFormActivity.java#L74 to a different status other that `RESULT_CANCELED`. If the issue is around permissions being rejected.

Reply:
> What toast are you referring to?

https://github.com/onaio/rdt-standard/blob/a546608c8d1fb5f101d3d175231cfb1af2afb716/rdt-open-reader/src/main/java/io/ona/rdt/activity/RDTJsonFormActivity.java#L75

Reply:
Even i changed the status but there is nothing have to do so, i think we remove `setResult(RESULT_CANCELED, null);` this

Reply:
does that work? what is the value of `resultCode` if we remove it?

Reply:
yes, the value of resultCode is `RESULT_CANCELED`

Reply:
what is the value if `setResult(RESULT_CANCELED, null);` was removed?

Reply:
if we removed the `setResult(RESULT_CANCELED, null);` then the value still `RESULT_CANCELED`

Reply:
then how would this fix the issue of crashing? wouldn't we still enter this else block?

Reply:
Actually, I just removed the explicit crash due to the permission failure Because we are already showing the toast in case the user denies permission. So, there is no need to throw an exception.

Reply:
My point was that the crash was not there for that purpose. We need to retain it to allow the app to close out completely in case a crash occurs from the jsonformactivity.

Currently, the only way I see to do that is to see if the result code of cancelled was returned.

Reply:
so if cancelled is returned when we remove the line, the we can use a custom code instead and leave the else block as is.

Reply:
fixed.